api/oneclick.py

```diff
@@ -206,14 +207,10 @@ def fulfill(self, patron, pin, licensepool, internal_format):
         :return a FulfillmentInfo object.
         """
 
-        patron_oneclick_id = self.validate_patron(patron)
+        patron_oneclick_id = self.patron_remote_identifier(patron)
         (item_oneclick_id, item_media) = self.validate_item(licensepool)
 
-        # find patron's checkouts
         checkouts_list = self.get_patron_checkouts(patron_id=patron_oneclick_id)
-        if not checkouts_list:
-            raise NoActiveLoan("Cannot fulfill %s - patron %s/%s has no checkouts.", item_oneclick_id, 
-                patron.authorization_identifier, patron_oneclick_id)
```

Comment:
This is unnecessary because if you have no checkouts, you don't have the checkout we're looking for, so you fall into the case below, and that error message is more helpful.